Source/Core/InputCommon/ControllerInterface/DualShockUDPClient/DualShockUDPClient.cpp

```diff
@@ -155,16 +154,40 @@ constexpr auto SERVER_LISTPORTS_INTERVAL = std::chrono::seconds{1};
 constexpr int TOUCH_X_AXIS_MAX = 1000;
 constexpr int TOUCH_Y_AXIS_MAX = 500;
 
-static bool s_server_enabled;
-static std::string s_server_address;
-static u16 s_server_port;
+struct Server
+{
+  Server(std::string description, std::string address, u16 port)
+      : m_description{description}, m_address{address}, m_port{port}
+  {
+  }
+  Server(const Server&) = delete;
+  Server(Server&& other)
```

Comment:
Any reason this isn't just a defaulted move constructor?

If it absolutely needs to be user-provided, this should also be marked noexcept.

Reply:
Due to there being a mutex, it couldn't be implicitly created.  The others are there because of rule of 5.  I'll mark it `noexcept` thanks!